File: app-android/src/main/kotlin/com/gpxeditor/android/screens/TrackDetailSections.kt
package com.gpxeditor.android.screens

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.gpxeditor.shared.feature.trackdetail.TrackSegmentSummary
import com.gpxeditor.shared.feature.trackdetail.TrackSummary

@Composable
fun SummarySection(
    title: String,
    importedAt: String?,
    summary: TrackSummary,
) {
    DetailSection(title = title) {
        importedAt?.let { DetailRow("Imported", it) }
        DetailRow("Tracks", summary.trackCount.toString())
        DetailRow("Segments", summary.segmentCount.toString())
        DetailRow("Points", summary.pointCount.toString())
        DetailRow("Distance", formatDistance(summary.distanceMeters))
        DetailRow("Elevation gain", formatElevation(summary.totalAscentMeters))
        DetailRow("Elevation loss", formatElevation(summary.totalDescentMeters))
        DetailRow("Elevation range",
            formatElevationRange(summary.minElevationMeters, summary.maxElevationMeters)
        )
        summary.averagePowerWatts?.let { DetailRow("Average power", "${it.toInt()} W") }
        summary.maxPowerWatts?.let { DetailRow("Maximum power", "$it W") }
        summary.averageCadenceRpm?.let { DetailRow("Average cadence", "${it.toInt()} rpm") }
        DetailRow("Time range", formatTimeRange(summary.startTime, summary.endTime))
        DetailRow("Start", formatCoordinate(summary.startCoordinate))
        DetailRow("Finish", formatCoordinate(summary.endCoordinate))
    }
}

@Composable
fun WarningsSection(warnings: List<String>) {
    if (warnings.isNotEmpty()) {
        DetailSection(title = "Warnings") {
            warnings.forEach { warning ->
                Text(
                    text = warning,
                    color = MaterialTheme.colorScheme.error,
                    style = MaterialTheme.typography.bodyMedium,
                )
            }
        }
    }
}

@Composable
fun TrackSegmentsSection(segments: List<TrackSegmentSummary>) {
    DetailSection(title = "Segments") {
        if (segments.isEmpty()) {
            Text(
                text = "No segments found.",
                style = MaterialTheme.typography.bodyMedium,
            )
        } else {
            segments.forEach { segment ->
                Column(verticalArrangement = Arrangement.spacedBy(4.dp)) {
                    Text(
                        text = "Segment ${segment.index}",
                        style = MaterialTheme.typography.titleSmall,
                    )
                    Text(
                        text = "${segment.pointCount} points / ${formatDistance(segment.distanceMeters)}",
                        style = MaterialTheme.typography.bodyMedium,
                    )
                    Text(
                        text = formatTimeRange(segment.startTime, segment.endTime),
                        style = MaterialTheme.typography.bodySmall,
                    )
                }
            }
        }
    }
}

@Composable
fun DetailSection(
    title: String,
    content: @Composable () -> Unit,
) {
    Surface(
        modifier = Modifier.fillMaxWidth(),
        tonalElevation = 1.dp,
        shape = MaterialTheme.shapes.small,
    ) {
        Column(
            modifier = Modifier.padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(10.dp),
        ) {
            Text(
                text = title,
                style = MaterialTheme.typography.titleMedium,
            )
            content()
        }
    }
}

@Composable
fun DetailRow(
    label: String,
    value: String,
) {
    Column(verticalArrangement = Arrangement.spacedBy(2.dp)) {
        Text(
            text = label,
            style = MaterialTheme.typography.labelMedium,
        )
        Text(
            text = value,
            style = MaterialTheme.typography.bodyMedium,
        )
    }
}

File: iosApp/GPXFixer/GPXFixer/screens/TrackDetailSections.swift
import SwiftUI
import Charts
import shared

struct PowerChartSection: View {
    let samples: [PowerChartSample]

    var body: some View {
        if samples.count >= 2 {
            Section("Power") {
                Chart(Array(samples.enumerated()), id: \.offset) { _, sample in
                    LineMark(
                        x: .value("Time", Double(sample.elapsedSeconds) / 60.0),
                        y: .value("Power", Int(sample.powerWatts))
                    )
                    AreaMark(
                        x: .value("Time", Double(sample.elapsedSeconds) / 60.0),
                        y: .value("Power", Int(sample.powerWatts))
                    )
                    .foregroundStyle(.linearGradient(
                        colors: [.accentColor.opacity(0.25), .accentColor.opacity(0.02)],
                        startPoint: .top,
                        endPoint: .bottom
                    ))
                }
                .chartXAxisLabel("Time, min")
                .chartYAxisLabel("Power, W")
                .frame(height: 180)
                .padding(.vertical, 8)
            }
        }
    }
}

struct WarningsSection: View {
    let warnings: [String]

    var body: some View {
        if !warnings.isEmpty {
            Section("Warnings") {
                ForEach(warnings, id: \.self) { warning in
                    Text(warning)
                        .foregroundStyle(.red)
                }
            }
        }
    }
}

struct TrackSegmentsSection: View {
    let segments: [TrackSegmentSummary]

    var body: some View {
        Section("Segments") {
            if segments.isEmpty {
                Text("No segments found.")
                    .foregroundStyle(.secondary)
            } else {
                ForEach(segments, id: \.index) { segment in
                    VStack(alignment: .leading, spacing: 6) {
                        Text("Segment \(segment.index)")
                            .font(.headline)
                        Text("\(segment.pointCount) points / \(formatDistance(segment.distanceMeters))")
                            .foregroundStyle(.secondary)
                        Text(formatTimeRange(segment.startTime, segment.endTime))
                            .font(.caption)
                            .foregroundStyle(.secondary)
                    }
                    .padding(.vertical, 4)
                }
            }
        }
    }
}

struct SummarySection: View {
    let title: String
    let importedAt: String?
    let summary: TrackSummary

    var body: some View {
        Section(title) {
            if let importedAt {
                DetailRow(label: "Imported", value: importedAt)
            }
            DetailRow(label: "Tracks", value: "\(summary.trackCount)")
            DetailRow(label: "Segments", value: "\(summary.segmentCount)")
            DetailRow(label: "Points", value: "\(summary.pointCount)")
            DetailRow(label: "Distance", value: formatDistance(summary.distanceMeters))
            DetailRow(label: "Elevation gain", value: formatElevation(summary.totalAscentMeters))
            DetailRow(label: "Elevation loss", value: formatElevation(summary.totalDescentMeters))
            DetailRow(
                label: "Elevation range",
                value: formatElevationRange(summary.minElevationMeters, summary.maxElevationMeters)
            )
            if let averagePower = summary.averagePowerWatts {
                DetailRow(label: "Average power", value: "\(Int(averagePower.doubleValue)) W")
            }
            if let maxPower = summary.maxPowerWatts {
                DetailRow(label: "Maximum power", value: "\(maxPower.intValue) W")
            }
            if let averageCadence = summary.averageCadenceRpm {
                DetailRow(label: "Average cadence", value: "\(Int(averageCadence.doubleValue)) rpm")
            }
            DetailRow(label: "Time range", value: formatTimeRange(summary.startTime, summary.endTime))
            DetailRow(label: "Start", value: formatCoordinate(summary.startCoordinate))
            DetailRow(label: "Finish", value: formatCoordinate(summary.endCoordinate))
        }
    }
}

struct DetailRow: View {
    let label: String
    let value: String

    var body: some View {
        VStack(alignment: .leading, spacing: 2) {
            Text(label)
                .font(.caption)
                .foregroundStyle(.secondary)
            Text(value)
        }
    }
}
